cmd/kubeadm/app/cmd/reset.go

```diff
@@ -127,7 +128,11 @@ func newResetData(cmd *cobra.Command, opts *resetOptions, in io.Reader, out io.W
 		klog.V(1).Infof("[reset] Could not obtain a client set from the kubeconfig file: %s", opts.kubeconfigPath)
 	}
 
-	ignorePreflightErrorsSet, err := validation.ValidateIgnorePreflightErrors(initCfg.FeatureGates, opts.ignorePreflightErrors, resetCfg.IgnorePreflightErrors)
+	featureList := make(map[string]bool)
+	if initCfg != nil {
+		featureList = initCfg.FeatureGates
+	}
```

Comment:
I'm a little confused. User should how to enable `MergeCLIArgumentsWithConfig` feature gate for `kubeadm reset`? It seems that the `featureList` will always be empty on worker nodes. Or I have missed something.

Reply:
I'm not sure if we handled feature gate correctly for `init/join/reset/upgrade`, we may need a double check for #119946 @chendave @pacoxu @neolit123

Reply:
If it's a worker node, the initCfg is nil, there is no way to tell if `MergeCLIArgumentsWithConfig` is enabled or not, the CLI override the config file is the default behavior.

Reply:
it's looks like to me this is a limitation of how FG is implemented in kubeadm, it's only applicable to control nodes, not just this FG but all other FG should be true as well.

Reply:
two choice here, 
- document this as a limitation, FG is now not configurable in the reset/join/init but only in the clusterCfg.
- or we need to revert https://github.com/kubernetes/kubernetes/pull/119946 and see if there is any other choice to implement it.

Reply:
I am fine with revert.

@neolit123 @pacoxu WDYT?

Reply:
+1